zkevm-circuits/src/evm_circuit/execution/push.rs

```diff
@@ -53,21 +56,16 @@ impl<F: Field> ExecutionGadget<F> for PushGadget<F> {
         //                           ▼                     ▼
         //   [byte31,     ...,     byte2,     byte1,     byte0]
         //
-        for idx in 0..32 {
+        for (idx, sel) in selectors.iter().enumerate() {
             let byte = &value.limbs[idx];
             let index = cb.curr.state.program_counter.expr() + opcode.expr()
-                - (OpcodeId::PUSH1.as_u8() - 1 + idx as u8).expr();
-            if idx == 0 {
+                - (OpcodeId::PUSH0.as_u8() + idx as u8).expr();
+
+            cb.condition(sel.expr(), |cb| {
```

Comment:
Shouldn't we also lookup-check the byte if is_push0 is true at idx=0 (we check later on that byte = 0 when is_push0 is true)?
In which case, for idx=0, we could remove the condition iiuc.